powersimdata/input/check.py

```diff
@@ -0,0 +1,195 @@
+    :raises ValueError: if any AC branches bridge across interconnections.
+    """
+    branch = grid.branch.copy()
+    bus = grid.bus
+    branch["from_interconnect"] = bus.loc[branch.from_bus_id, "interconnect"].tolist()
+    branch["to_interconnect"] = bus.loc[branch.to_bus_id, "interconnect"].tolist()
+    interconnect_match = branch["from_interconnect"] == branch["to_interconnect"]
+    if not interconnect_match.all():
+        non_matching_branches = interconnect_match.index[~interconnect_match]
+        raise ValueError(
+            "Branch(es) connected across multiple interconnections: "
+            f"{non_matching_branches}."
+        )
+
+
+def _check_transformer_substations(grid):
+    """Check whether any transformers are are not within exactly one same substation.
+
+    :param powersimdata.input.grid.Grid grid: grid or grid-like object to check.
+    :raises ValueError: if any transformers are not within exactly one same substation.
+    """
+    txfmr_branch_types = {"Transformer", "TransformerWinding"}
+    branch = grid.branch
+    bus2sub = grid.bus2sub
+    transformers = branch.loc[branch.branch_device_type.isin(txfmr_branch_types)].copy()
+    transformers["from_sub"] = bus2sub.loc[transformers.from_bus_id, "sub_id"].tolist()
+    transformers["to_sub"] = bus2sub.loc[transformers.to_bus_id, "sub_id"].tolist()
+    transformer_sub_match = transformers["from_sub"] == transformers["to_sub"]
+    if not transformer_sub_match.all():
+        non_matching_transformers = transformer_sub_match.index[~transformer_sub_match]
+        raise ValueError(
+            "Transformer(s) connected across multiple substations: "
+            f"{non_matching_transformers}."
+        )
+
+
+def _check_line_voltages(grid):
+    """Check whether any lines connect across difference voltage levels.
+
+    :param powersimdata.input.grid.Grid grid: grid or grid-like object to check.
+    :raises ValueError: if any lines connect across difference voltage levels.
+    """
+    lines = grid.branch.query("branch_device_type == 'Line'").copy()
+    lines["from_kV"] = grid.bus.loc[lines.from_bus_id, "baseKV"].tolist()
+    lines["to_kV"] = grid.bus.loc[lines.to_bus_id, "baseKV"].tolist()
+    line_voltage_match = lines["from_kV"] == lines["to_kV"]
+    if not line_voltage_match.all():
+        non_matching_voltages = line_voltage_match.index[~line_voltage_match]
+        raise ValueError(
+            f"Lines(s) connected across multiple voltages: {non_matching_voltages}."
+        )
+
+
+def _check_plant_against_gencost(grid):
+    """Check whether indices of plant and gencost tables match.
+
+    :param powersimdata.input.grid.Grid grid: grid or grid-like object to check.
+    :raises ValueError: if indices of plant and gencost tables don't match.
+    """
+    if not (
+        set(grid.plant.index) == set(grid.gencost["before"].index)
+        and set(grid.plant.index) == set(grid.gencost["after"].index)
```

Comment:
I think we could chain the comparison here, e.g. `if not a == b == c: ...`

Reply:
Done.